Reviewing getting started guides (#10979)
DataDog/documentation: content/en/getting_started/synthetics/api_test.md

    - link: '/synthetics/identify_synthetics_bots'
      tag: 'Documentation'
      text: 'Learn how to identify Synthetic bots for API tests'
    - link: '/api/v1/synthetics/#create-a-test'
      tag: 'API Docs'
      text: 'Create a Synthetic test programmatically'

---

## Create a single API test

> [API tests][1] **proactively monitor** that your most important services are available at anytime and from anywhere. API tests come in four different flavors that allow you to launch requests on the **different network layers** of your systems.

Reviewer: ```suggestion
[API tests][1] **proactively monitor** that your most important services are available at anytime and from anywhere. API tests come in four different subtypes that allow you to launch requests on the **different network layers** of your systems.
```

The example below demonstrates the creation of an [HTTP test][2], a subtype of [API tests][1]. [HTTP tests][1] **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your applications are responding to requests and meeting any conditions you define, such as expected **response time**, **HTTP status code**, and **header or body contents**.

### Define request

1. In the Datadog application, hover over **[UX Monitoring][3]** in the left hand menu and select **[Synthetic Tests][3]**.
2. In the top right corner, click the **New Test** button.
3. Select **[New API test][4]**.
4. Select the `HTTP` request type.
5. Define your request:

    - Add the URL of the endpoint you want to monitor. If you don’t know what to start with, you can use `https://www.shopist.io/`, a test web application. Defining the endpoint to test automatically populates the name of your test to `Test on www.shopist.io`. You can change your test name to something else if you want to.
    - You can select **Advanced Options** to use custom request headers, authentication credentials, body content, or cookies.

>     - You can set tags such as `env:prod` and `app:shopist` on your test. Tags allow you to keep your test suite organized and quickly find test you're interested in on the homepage.

Reviewer: ```suggestion
    - You can set tags such as `env:prod` and `app:shopist` on your test. Tags allow you to keep your test suite organized and quickly find tests you're interested in on the homepage.
```

### Define assertions

Clicking **Test URL** automatically populates basic assertions about your endpoint's response. Assertions define what a successful test run is.

In this example, three default assertions populate after triggering the sample test run:

{{< img src="getting_started/synthetics/assertions-example.png" alt="Browser test failure"  style="width:90%;">}}

Assertions are fully customizable. To add a custom assertion, click on elements of the response preview or click **New Assertion**. 

{{< img src="getting_started/synthetics/api-test-configuration-2.gif" alt="Browser test failure"  style="width:90%;">}}

### Select locations 

Select one or more **Managed Locations** or **Private Locations** to run your test from.

Managed locations allow you to test public-facing websites and endpoints. To test internal applications or simulate user behavior in discrete geographic regions, select one of your **Private Locations** instead.

For more information on how to set up private locations, see [Getting Started with Private Locations][5].

### Specify test frequency

Select the frequency of your test runs.

### Define alert conditions

> You can define alert conditions to ensure your test does not trigger in case of sporadic network blip and that you only get alerted in case of real issues with your endpoint.

Reviewer: ```suggestion
You can define alert conditions to ensure your test does not trigger for things like a sporadic network blip, so that you only get alerted in case of real issues with your endpoint.
```

You can specify the number of consecutive failures that should happen before considering a location failed:

```text
Retry test 2 times after 300 ms in case of failure
```

> You can also decide to only trigger a notification when your endpoint goes down for a certain amount of time and number of locations. In the below example, we set our alerting rule to send a notification if our test fails for three minutes on two different locations:

Reviewer: ```suggestion
You can also define it to only trigger a notification when your endpoint goes down for a certain amount of time and number of locations. In the below example, the alerting rule is set to send a notification if the test fails for three minutes on two different locations:
```
